packages/core/src/analyticsConnector.ts

```diff
@@ -0,0 +1,35 @@
+export type AnalyticsEvent = {
+  eventType: string;
+  eventProperties?: Record<string, unknown>;
+  userProperties?: Record<string, unknown>;
+};
+
+export type AnalyticsEventReceiver = (event: AnalyticsEvent) => void;
+
+export interface AnalyticsConnector {
+  logEvent(event: AnalyticsEvent): void;
+  setEventReceiver(listener: AnalyticsEventReceiver): void;
+}
+
+export class AnalyticsConnectorImpl implements AnalyticsConnector {
+  private receiver: AnalyticsEventReceiver;
+  private queue: AnalyticsEvent[] = [];
+
+  logEvent(event: AnalyticsEvent): void {
+    if (!this.receiver) {
+      this.queue.push(event);
```

Comment:
The queue could grow unbounded if no receiver is added. Maybe we could add a max queue size or user warning is no receiver is set.